### PR TITLE
Update CMakeLists.txt

### PR DESCRIPTION
**Pull request by haonanya**
1. Add additional link dependencies for NVPTX and AMDGPU
2. Create stripped pdbs for Windows
3. Remove windows_resource_file in windows

### CODE REVIEW

File: common_clang.cpp [hunk 1]
```diff
@@ -319,7 +319,7 @@ Compile(const char *pszProgramSource, const char **pInputHeaders,
         if (pBinaryResult) {
           *pBinaryResult = nullptr;
         }
-        assert(!"Failed to read just compiled LLVM IR!");
+        assert(0 && "Failed to read just compiled LLVM IR!");
```
Comment (hewj03) [inline common_clang.cpp:322]:
```suggestion
        assert(false && "Failed to read just compiled LLVM IR!");
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -85,6 +85,7 @@ if (NOT WIN32)
 endif()
 
 use_rtti(FALSE)
+use_eh(TRUE)
```
Comment (hewj03) [inline CMakeLists.txt:88]: we only need to enable exception on in opencl-clang tests

**Review by hewj03 — APPROVED** [review]
LGTM. Please consider removing use_eh from library files and investigating if we can change add_library back to add_llvm_library. I suggest to create issues to track these two tasks.

**Comment by haonanya** [comment]
> LGTM. Please consider removing use_eh from library files and investigating if we can change add_library back to add_llvm_library. I suggest to create issues to track these two tasks.

Thanks. File https://github.com/intel/opencl-clang/issues/400 and https://github.com/intel/opencl-clang/issues/401 to track.

**Pull request merged**